openmc/filter.py

```diff
@@ -0,0 +1,1810 @@
+from abc import ABCMeta
+from collections import OrderedDict
+from collections.abc import Iterable
+import copy
+import hashlib
+from itertools import product
+from numbers import Real, Integral
+from xml.etree import ElementTree as ET
+
+import numpy as np
+import pandas as pd
+
+import openmc
+import openmc.checkvalue as cv
+from .cell import Cell
+from .material import Material
+from .mixin import IDManagerMixin
+from .surface import Surface
+from .universe import Universe
+
+
+_FILTER_TYPES = (
+    'universe', 'material', 'cell', 'cellborn', 'surface', 'mesh', 'energy',
+    'energyout', 'mu', 'polar', 'azimuthal', 'distribcell', 'delayedgroup',
+    'energyfunction', 'cellfrom', 'legendre', 'spatiallegendre',
+    'sphericalharmonics', 'zernike', 'zernikeradial', 'particle'
+)
+
+_CURRENT_NAMES = (
+    'x-min out', 'x-min in', 'x-max out', 'x-max in',
+    'y-min out', 'y-min in', 'y-max out', 'y-max in',
+    'z-min out', 'z-min in', 'z-max out', 'z-max in'
+)
+
+_PARTICLES = {'neutron', 'photon', 'electron', 'positron'}
+
+
+class FilterMeta(ABCMeta):
+    def __new__(cls, name, bases, namespace, **kwargs):
+        # Check the class name.
+        if not name.endswith('Filter'):
```

Comment:
I would recommend move 'Filter' to a variable for use here and in subsequent lines to allow for reusability and avoid errors due to typos.
`const REQUIRED_SUFFIX = 'Filter'`